index.js

```diff
@@ -9,12 +9,28 @@
   // This will be replaced by native code on demand.
   const siteSpecificDetector = null
 
-  function fixUpRelativeUrl (url) {
+  function isHttpsScheme (url) {
     if (!url || typeof url !== 'string') {
-      return url
+      return false
     }
 
-    if (!url.startsWith('http://') || !url.startsWith('https://')) {
+    let isHttpsScheme = false
+    try {
+      // In case of http: or data: protocol, the base URL is not used
+      isHttpsScheme = new URL(url, window.location.origin).protocol === 'https:'
+    } catch (e) {
+      // Ignore
+    }
+
+    return isHttpsScheme
+  }
+
+  function fixUpUrl (url) {
+    if (!isHttpsScheme(url)) {
+      return null
+    }
+    
+    if (!url.startsWith('https://')) {
```

Comment:
will this block ever be reached given that you return early if the URL is not HTTPS? https://github.com/brave/playlist-component/pull/33/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R29

Reply:
Yes, I believe it's reachable when it's relative url.